roxie/ccd/ccdquery.cpp

```diff
@@ -515,6 +524,9 @@ class CQueryFactory : public CInterface, implements IQueryFactory, implements IR
         if (!helperFactory)
             throw MakeStringException(ROXIE_INTERNAL_ERROR, "Internal error: helper function %s not exported", helperName.str());
 
+        RemoteActivityId remoteId(rid, hashValue);
+        RemoteActivityId remoteId2(rid | ROXIE_ACTIVITY_FETCH, hashValue);
+
```

Comment:
It could do.  The overhead is tiny, and the code is not time critical, so I would take it as it stands.